src/nvim/cursor_shape.c

```diff
@@ -36,11 +38,61 @@ static cursorentry_T shape_table[SHAPE_IDX_COUNT] =
   {0, 0, 0, 100L, 100L, 100L, 0, 0, "sm", SHAPE_CURSOR},
 };
 
-/*
- * Parse the 'guicursor' option ("what" is SHAPE_CURSOR) or 'mouseshape'
- * ("what" is SHAPE_MOUSE).
- * Returns error message for an illegal option, NULL otherwise.
- */
+
+Dictionary cursor_shape_dict(void)
+{
+  /* Array args = ARRAY_DICT_INIT; */
+  Dictionary all = ARRAY_DICT_INIT;
+  char buf[30] = "uninitialized";
+
+  /* shape_table */
+  for (int i = 0; i < SHAPE_IDX_COUNT; i++) {
+
+    ILOG("Adding entry");
+    Dictionary dic = ARRAY_DICT_INIT;
+    cursorentry_T *cur = &shape_table[i];
+    /* I noticed msgp*/
+    /* switch (cur->shape) { */
+    /*   case SHAPE_BLOCK: STRCPY(buf, "block"); */
+    /*   case SHAPE_VER: STRCPY(buf, "vertical"); */
+    /*   case SHAPE_HOR: STRCPY(buf, "horizontal"); */
+    /*   default:; */
+    /* }; */
+    /* PUT(dic, "shape", STRING_OBJ(buf)); */
```

Comment:
Yes. It's more a matter of not wanting to expose the internal order to the API, and then having to support those numbers forever more. It also is hopefully a better experience for API users.